From f74f3edb8dc65646c3d30a3b5280a6ba284f26a1 Mon Sep 17 00:00:00 2001
From: Ankit Jain <ankit-aj.jain@broadcom.com>
Date: Mon, 23 Jun 2025 05:54:28 +0000
Subject: [PATCH] Disable default composefs enablement in postprocess

- We don't enable composefs as it require
 to support for immutable filesystem layer
- composefs is an experimental feature
- In postprocess, it tries to add metadata
 for composefs but failed because it doesn't support
- Hence disabling the composefs in postprocess
 which basically reverts following commits:
https://github.com/coreos/rpm-ostree/commit/e2103c5643ccc454b3ec0c5b52bf48f45966c891
https://github.com/coreos/rpm-ostree/commit/4b1e9567ae895c18b07ec62be20a91cce3038f33

Signed-off-by: Ankit Jain <ankit-aj.jain@broadcom.com>
---
 src/libpriv/rpmostree-postprocess.cxx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libpriv/rpmostree-postprocess.cxx b/src/libpriv/rpmostree-postprocess.cxx
index 6d0db1be..0458d97b 100644
--- a/src/libpriv/rpmostree-postprocess.cxx
+++ b/src/libpriv/rpmostree-postprocess.cxx
@@ -884,11 +884,11 @@ rpmostree_compose_commit (int rootfs_fd, OstreeRepo *repo, const char *parent_re
 
   // Unfortunately these API takes GVariantDict, not GVariantBuilder, so convert
   g_autoptr (GVariantDict) metadata_dict = g_variant_dict_new (src_metadata);
-
+#if 0
   if (!ostree_repo_commit_add_composefs_metadata (repo, 0, metadata_dict,
                                                   (OstreeRepoFile *)root_tree, cancellable, error))
     return glnx_prefix_error (error, "Adding composefs metadata");
-
+#endif
   if (!container)
     {
       if (!ostree_commit_metadata_for_bootable (root_tree, metadata_dict, cancellable, error))
-- 
2.39.4

